hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/metrics/MetricsOperator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core.metrics;
+
+/**
+ * Marker interface to be implemented by all classes that want to be able
+ * to retrieve and work with custom metrics.
+ */
+public interface MetricsOperator {
```

Comment:
could we find a different name? What does MetricsOperator mean?

Reply:
`MetricsOperator` is supposed to be an/any entity which can set the value of user metrics; that's why I'm trying names like `MetricsSource`, `MetricsSetter`, `MetricsOperator` and such, none of them particularly good.. will try to find a better name.

Reply:
`MetricsGenerator`?

Reply:
something like HasMetrics or ProvidesMetrics